src/ZenohUTransport.cpp

```diff
@@ -9,4 +9,384 @@
+	UStatus status;
+	status.set_code(code);
+	status.set_message(std::string(message));
+	return status;
+}
+
+std::string ZenohUTransport::toZenohKeyString(
+    const std::string& default_authority_name, const UUri& source,
+    const std::optional<v1::UUri>& sink) {
+	std::ostringstream zenoh_key;
+
+	auto writeUUri = [&](const v1::UUri& uuri) {
+		zenoh_key << "/";
+
+		// authority_name
+		if (uuri.authority_name().empty()) {
+			zenoh_key << default_authority_name;
+		} else {
+			zenoh_key << uuri.authority_name();
+		}
+		zenoh_key << "/";
+
+		// ue_id
+		if (uuri.ue_id() == WILDCARD_ENTITY_ID) {
+			zenoh_key << "*";
+		} else {
+			zenoh_key << uuri.ue_id();
+		}
+		zenoh_key << "/";
+
+		// ue_version_major
+		if (uuri.ue_version_major() == WILDCARD_ENTITY_VERSION) {
+			zenoh_key << "*";
+		} else {
+			zenoh_key << uuri.ue_version_major();
+		}
+		zenoh_key << "/";
+
+		// resource_id
+		if (uuri.resource_id() == WILDCARD_RESOURCE_ID) {
+			zenoh_key << "*";
+		} else {
+			zenoh_key << uuri.resource_id();
+		}
+	};
+
+	zenoh_key << "up";
+	zenoh_key << std::uppercase << std::hex;
+
+	writeUUri(source);
+
+	if (sink.has_value()) {
+		writeUUri(*sink);
+	} else {
+		zenoh_key << "/{}/{}/{}/{}";
+	}
+	return zenoh_key.str();
+}
+
+std::vector<std::pair<std::string, std::string>>
+ZenohUTransport::uattributesToAttachment(const UAttributes& attributes) {
+	std::vector<std::pair<std::string, std::string>> res;
+
+	std::string version(&UATTRIBUTE_VERSION, 1);
+
+	std::string data;
+	attributes.SerializeToString(&data);
+
+	res.push_back(std::make_pair("", version));
+	res.push_back(std::make_pair("", data));
+	return res;
+}
+
+UAttributes ZenohUTransport::attachmentToUAttributes(
+    const AttachmentView& attachment) {
+	std::vector<BytesView> attachment_vec;
+	attachment.iterate(
+	    [&](const BytesView& key, const BytesView& value) -> bool {
+		    attachment_vec.push_back(value);
+		    return true;
+	    });
+
+	if (attachment_vec.size() != 2) {
+		// TODO: error report, exception?
+	}
+
+	if (attachment_vec[0].get_len() == 1) {
+		if (attachment_vec[0].as_string_view()[0] != UATTRIBUTE_VERSION) {
+			// TODO: error report, exception?
+		}
+	};
+	UAttributes res;
+	// TODO: more efficient way?
+	res.ParseFromString(std::string(attachment_vec[1].as_string_view()));
+	return res;
+}
+
+Priority ZenohUTransport::mapZenohPriority(UPriority upriority) {
+	switch (upriority) {
+		case UPriority::UPRIORITY_CS0:
+			return Z_PRIORITY_BACKGROUND;
+		case UPriority::UPRIORITY_CS1:
+			return Z_PRIORITY_DATA_LOW;
+		case UPriority::UPRIORITY_CS2:
+			return Z_PRIORITY_DATA;
+		case UPriority::UPRIORITY_CS3:
+			return Z_PRIORITY_DATA_HIGH;
+		case UPriority::UPRIORITY_CS4:
+			return Z_PRIORITY_INTERACTIVE_LOW;
+		case UPriority::UPRIORITY_CS5:
+			return Z_PRIORITY_INTERACTIVE_HIGH;
+		case UPriority::UPRIORITY_CS6:
+			return Z_PRIORITY_REAL_TIME;
+		case UPriority::UPRIORITY_UNSPECIFIED:
+		default:
```

Comment:
Remove `default:` - we will turn on warnings as errors. Then the compiler will fail to compile if a new value is added to the proto files but this switch is not updated.

Reply:
Check gcc documentation - it may be possible to have `default:` and still get the warning with the right flags set.

Reply:
We have the required flags. I'll close this request